app/medInria/medFilteringWorkspace.cpp

```diff
@@ -97,67 +78,60 @@ void medFilteringWorkspace::setupViewContainerStack()
 
 void medFilteringWorkspace::changeToolBoxInput()
 {
-    if(d->filteringToolBox.isNull())
-        return;
-
-    if(!d->inputContainer->view())
+    if(!selectorToolBox().isNull())
     {
-        d->filteringToolBox->clear();
-        return;
-    }
-
-    medAbstractLayeredView *layeredView = dynamic_cast<medAbstractLayeredView *>(d->inputContainer->view());
-    if(!layeredView)
-    {
-        qWarning() << "Non layered view are not supported in filtering workspace yet.";
-        d->filteringToolBox->clear();
-        return;
+        if(!d->inputContainer->view() && (selectorToolBox()->data() != NULL))
+        {
+            selectorToolBox()->clear();
+        }
+        else
+        {
+            medAbstractLayeredView *layeredView = dynamic_cast<medAbstractLayeredView *>(d->inputContainer->view());
+            selectorToolBox()->onInputSelected(layeredView->layerData(layeredView->currentLayer()));
+        }
     }
-    d->filteringToolBox->onInputSelected(layeredView->layerData(layeredView->currentLayer()));
 }
 
 /**
  * @brief adds metadata to the output and emits a signal outputDataChanged(medAbstractData *)
  */
 void medFilteringWorkspace::onProcessSuccess()
 {
-    if(d->filteringToolBox.isNull())
-        return;
-
-    d->filterOutput = d->filteringToolBox->currentToolBox()->processOutput();
-    if ( !d->filterOutput )
-        return;
-
-    qDebug() << "d->filterOutput->identifier()" << d->filterOutput->identifier();
-
-    medAbstractData *inputData(d->filteringToolBox->data());
+    if(selectorToolBox()->data())
```

Comment:
As we use `selectorToolBox()->data()` lower, it can be interesting to test it. I can look at it indeed.